core/services/ardupilot_manager/flight_controller_detector/linux/linux_boards.py

```diff
@@ -0,0 +1,78 @@
+import os
+from typing import List, Optional, Type
+
+from loguru import logger
+from smbus2 import SMBus
+
+from typedefs import FlightController, PlatformType, Serial
+
+
+class LinuxFlightController(FlightController):
+    """Linux-based Flight-controller board."""
+
+    # for sanity reasons, let's assume a linux board never gets disconnected
+    # this will prevent a lot of loading/unloading of modules and overlays in the future
+    previously_detected: Optional["LinuxFlightController"] = None
+
+    @property
+    def type(self) -> PlatformType:
+        return PlatformType.Linux
+
+    def detect(self) -> bool:
+        raise NotImplementedError
+
+    def get_serial_cmdlines(self) -> str:
+        cmdlines = [f"-{entry.port} {entry.endpoint}" for entry in self.get_serials()]
+        return " ".join(cmdlines)
+
+    def get_serials(self) -> List[Serial]:
+        raise NotImplementedError
+
+    def check_for_i2c_device(self, bus_number: int, address: int) -> bool:
+        try:
+            bus = SMBus(bus_number)
+            bus.read_byte_data(address, 0)
+            return True
+        except OSError:
+            return False
+
+    @classmethod
+    def get_all_subclasses(cls) -> List[Type["LinuxFlightController"]]:
+        all_subclasses = []
+
+        for subclass in cls.__subclasses__():
+            all_subclasses.append(subclass)
+            all_subclasses.extend(subclass.get_all_subclasses())
+
+        return all_subclasses
+
+    @staticmethod
+    def load_subclasses() -> None:
+        current_directory = os.path.dirname(__file__)
+
+        module_files = [
+            os.path.basename(filename)[:-3]  # Remove the '.py' extension
+            for filename in os.listdir(current_directory)
+            if filename.endswith(".py")
+            and not filename.endswith("__init__.py")
+            and not filename.endswith("linux_boards.py")
+        ]
+
+        for module in module_files:
+            logger.info(f"Loading module: {module}")
+            __import__(f"flight_controller_detector.linux.{module}")
```

Comment:
You should be able to load by just importing the files, check AbstractRouter from ardupilot manager: https://github.com/bluerobotics/BlueOS/blob/bd8069d45e6d816ab83ee5d928dabfb4236e9cef/core/services/ardupilot_manager/mavlink_proxy/Manager.py#L8

Reply:
the idea was that all files dropped here would be loade automagically, but it is a lot of code.
One issue is that importing them here causes a cyclic redundancy, so they (or at least argonot) would need to be imported within the function here

Reply:
Can't be imported as the others and keep the subclass for subclass logic ?

Reply:
actually I think I have a solution for that